modules/@apostrophecms/schema/ui/apos/components/AposInputSlug.vue

```diff
@@ -81,15 +81,23 @@ export default {
     }
   },
   watch: {
-    followingValue(newValue, oldValue) {
-      if (this.compatible(oldValue, this.next)) {
-        // If this is a page slug, we only replace the last section of the slug.
-        if (this.field.page) {
-          const parts = this.next.match(/[^/]+/g);
-          parts.pop();
-          this.next = `/${parts.join('/')}${this.slugify(newValue)}`;
-        } else {
-          this.next = this.slugify(newValue);
+    followingValues: {
+      // We are usually interested in followingValue.title, but a
+      // secondary slug field could be configured to watch
+      // one or more other fields
+      deep: true,
+      handler(newValue, oldValue) {
+        oldValue = Object.values(oldValue).join(' ');
+        newValue = Object.values(newValue).join(' ');
```

Comment:
Should we bail of the values are objects or arrays?

Reply:
The only values we will ever receive are those we are `following`. This input field type should be documented as having support for `following` fields that convert themselves sensibly to a string. I don't think we need to do more than that.

Reply:
(Documentation being a task we all know is forthcoming.)